src/engraving/dom/staff.cpp

```diff
@@ -1040,17 +1042,351 @@ const CapoParams& Staff::capo(const Fraction& tick) const
         return dummy;
     }
     --it;
-    return m_capoMap.at(*it);
+    return m_capoMap.at(*it).params;
 }
 
-void Staff::insertCapoParams(const Fraction& tick, const CapoParams& params)
+void Staff::applyCapoTranspose(int startTick, int endTick, const CapoParams& params, int notePitchCorrection /* = 0 */)
```

Comment:
This function is very hard to read. A lot is going on in it, with a lot of nested code. It would be better to split it into several smaller functions